EIPS/eip-ens-as-holder

```diff
@@ -0,0 +1,74 @@
+---
+eip: <to be assigned>
+title: ENS as Token Holder
+description: An interface for holding tokens by ENS.
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category ERC
+created: 2022-07-12
+requires: 137
+---
+
+## Abstract
+An interface for smart contract to become a holder of tokens by honoring ENS ownership.
+This specification supports EIP-721 and EIP-1155 tokens and is expandable to future token standard or other standard.
+
+## Motivation
+Currently, if someone wants to receive token, they have to setup a wallet address, otherwise,
+the minter or transferer will not be able to transfer or mint for that token. 
+
+By creating smart contract that conform to this interface specification, we can decouple
+the transfer of ownership from the existence of wallet addresses.
```

Comment:
```suggestion
Currently, if someone wants to receive a token, they have to set up a wallet address. By creating a smart contract that conforms to this interface specification, it is possible to decouple the transfer of ownership from the existence of wallet addresses.
```

One caveat: how do you expect one to set up an ENS address without first setting up a wallet and giving it ether?

Reply:
Such ENS can be *presumed* to exist. For example, one day Github as a company decided to adopt ENS, and hence they obtained the `github.eth` and  *guarantee* they will assign the `<user_id>.github.eth` to the user of them by validating their id.

In such case, a Token creator can airdrop their token to the `Pandapip1.github.eth` and Pandapip1 can *later* decide to connect wallet and authenticate with Github to claim the token being aidropped to this ENS.

It's true that *when* Pandapip1 setup its ENS, it will require a wallet address, but as demonstrated above, the token creator can conduct the token transfer/airdrop before the receipient set up it.